suite-common/wallet-utils/src/exportTransactions.ts

```diff
@@ -128,62 +136,112 @@ const prepareContent = (data: Data): Fields[] => {
                         other: '',
                     };
+                    feeUsed = true;
+                    return tokenData;
                 });
             }
-            return t.targets.map((target, index) => {
-                if (!target?.addresses?.length || !target?.amount) {
-                    return null;
-                }
-                return {
-                    ...sharedData,
-                    fee: index === 0 ? t.fee : '', // fee only once per tx
-                    feeSymbol: index === 0 ? coin.toUpperCase() : '',
-                    address: target.isAddress ? target.addresses[0] : '', // SENT - it is destination address, RECV - it is MY address
-                    label: target.isAddress && target.metadataLabel ? target.metadataLabel : '',
-                    amount: target.isAddress ? target.amount : '',
-                    symbol: target.isAddress ? coin.toUpperCase() : '',
-                    fiat:
-                        target.isAddress && target.amount && t.rates && t.rates[data.localCurrency]
-                            ? localizeNumber(
-                                  new BigNumber(target.amount)
-                                      .multipliedBy(t.rates[data.localCurrency]!)
-                                      .toNumber(),
-                                  undefined,
-                                  2,
-                                  2,
-                              ).toString()
-                            : '',
-                    other: !target.isAddress ? target.addresses[0] : '', // e.g. OP_RETURN
-                };
-            });
+
+            if (t.internalTransfers.length > 0) {
+                internalTransfers = t.internalTransfers.map(internal => {
+                    const internalTransferData = {
+                        ...sharedData,
+                        fee: !feeUsed ? t.fee : '', // fee only once per tx
+                        feeSymbol: !feeUsed ? coin.toUpperCase() : '',
+                        address: internal.to || '', // SENT - it is destination address, RECV - it is MY address
+                        label: '', // internal transactions do not have labels
+                        amount: internal.amount,
+                        symbol: coin.toUpperCase(), // if symbol not available, use contract address
+                        fiat:
+                            internal.amount && t.rates && t.rates[data.localCurrency]
+                                ? localizeNumber(
+                                      new BigNumber(internal.amount)
+                                          .multipliedBy(t.rates[data.localCurrency]!)
+                                          .toNumber(),
+                                      undefined,
+                                      2,
+                                      2,
+                                  ).toString()
+                                : '',
```

Comment:
same here

Reply:
I dont want to touch export transactions much. I have it in followup to totally change the logic.